### PR TITLE
Make `-with-past` the default in the ONNX export

### PR DESCRIPTION
**Pull request by fxmarty**
As per title.

Breaking change: with-past is now the default.

### CODE REVIEW

File: optimum/exporters/onnx/__main__.py [hunk 1]
```diff
@@ -68,6 +68,17 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
+    # make -with-past the default if --task was not explicitely specified
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/__main__.py:71]:
```suggestion
    # Make -with-past the default if --task was not explicitely specified
```

**Review by michaelbenayoun — COMMENTED** [review]
Thankls for taking care of that!

I think we should also log a warning when a user could use past and don't (because he might not be aware of it). WDTY?

**Comment by fxmarty** [comment]
> I think we should also log a warning when a user could use past and don't (because he might not be aware of it). WDTY?

I don't think it is necessary as it will be the case only if the user explicitly pass `--task`

File: docs/source/exporters/onnx/usage_guides/export_a_model.mdx [hunk 2]
```diff
@@ -198,7 +198,7 @@ In the ONNX export, the past keys/values are reused by default. This behavior co
 A model exported using past key/values can be reused directly into Optimum's [`~onnxruntime.ORTModel`]:
 
 ```bash
-optimum-cli export onnx --model gpt2 --for-ort --task causal-lm-with-past gpt2_onnx/
+optimum-cli export onnx --model gpt2 --for-ort gpt2_onnx/
```
[the remark this answers is not in the document]
Comment (fxmarty) [reply docs/source/exporters/onnx/usage_guides/export_a_model.mdx:201]: I think so, yes

**Comment by michaelbenayoun** [comment]
Which is mandatory to do for for local models. I think having this warning will let the user know that it exists and it should be used whenever possible.

**Comment by fxmarty** [comment]
Makes sense!

File: optimum/exporters/onnx/__main__.py [hunk 3]
```diff
@@ -74,6 +74,20 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
+    if task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
+        model.config.model_type.replace("_", "-"), "onnx"
+    ):
+        if args.task == "auto":  # Make -with-past the default if --task was not explicitely specified
+            task = task + "-with-past"
+        elif args.task in ["seq2seq-lm", "causal-lm", "speech2seq-lm", "vision2seq-lm"]:
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/__main__.py:82]: I think we can infer task that can have a `with-past` flavor dynamically here.

**Review by michaelbenayoun — APPROVED** [review]
LGTM once my comment as been discussed / addressed!

File: optimum/exporters/onnx/__main__.py (hunk 3 above)
Comment (fxmarty) [reply optimum/exporters/onnx/__main__.py]: fixed thanks

**Review by michaelbenayoun — APPROVED** [review]

**Pull request merged**